OrbitCore/SamplingProfiler.h

```diff
@@ -111,6 +107,22 @@ class SamplingProfiler {
   [[nodiscard]] const ThreadSampleData* GetSummary() const;
   [[nodiscard]] uint32_t GetCountOfFunction(uint64_t function_address) const;
 
+  // TODO(irinashkviro): remove this when move to protobuf
+  void SaveCallstacks() {
```

Comment:
What's the difference between `m_Callstacks` and `callstacks_vector`, other than the former uses `BlockChain`, which is probably unnecessary? Could we just use one and then we don't need these two methods, which are actually a bit confusing?

Reply:
agree - I do not think blockchain can be replaced with vector.

Reply:
The `callstacks_vector` was introduced as a temporary solution and removed in the next PR (#865 ) with these 2 methods. Cereal don't know how to serialize blockchain until I write how to do it, but serialization is easy with the vector. If we don't want to replace blockchain with the vector, having a temporary field just for serialization is the easiest solution. Another option is to add cereal to blockchain, but I preferred to localize the serialization logic in SamplingProfiler in order to remove it in the next PR. Or am I missed something?

Reply:
There are ways to do that without an intermediate vector, but not with the ORBIT_SERIALIZE_... macros.
Out-of-class serialization functions for BlockChain are also possible, but also more work than needed for a temporary fix. I would stick to this one.